hist/histpainter/src/TGraphPainter.cxx

```diff
@@ -2567,89 +2613,74 @@ void TGraphPainter::PaintGraphAsymmErrors(TGraph *theGraph, Option_t *option)
       if (option3) {
          xline[if1-1] = x;
          xline[if2-1] = x;
-         yline[if1-1] = gPad->YtoPad(theY[i] + theEYhigh[i]);
-         yline[if2-1] = gPad->YtoPad(theY[i] - theEYlow[i]);
+         yline[if1-1] = yup2;
+         yline[if2-1] = ylow2;
          if1++;
          if2--;
          continue;
       }
 
-      if (xl1 > xl2) {
+      if (exl || exh) {
```

Comment:
Why are you using the double variables `exl` and `exh` as booleans  ? Should be  `exl > exh` ?